### PR TITLE
Added support for application crash logging

### PR DESCRIPTION
**Pull request by Pho3niX90**
Solves : https://github.com/pires/android-obd-reader/issues/135

Just replace the email with a correct one. on line 281

### CODE REVIEW

**Comment by sytolk** [comment]
no comment :)

**Comment by Pho3niX90** [comment]
@sytolk What exactly more do you need than a complete logcat + device information + android info?

**Comment by sytolk** [comment]
"complete logcat" - I don`t want it (I want logcat filtered by application PID)
I think that this email is exactly what you need to debug some application:
https://github.com/d-a-n/acra-mailer#sample-e-mail-error-report

**Comment by Pho3niX90** [comment]
@sytolk Okay let me rephrase myself. This gives you a complete log for the application, that includes debug message, error messages and info messages.

File: src/main/java/com/github/pires/obd/reader/io/ObdGatewayService.java [hunk 1]
```diff
@@ -274,5 +274,38 @@ public ObdGatewayService getService() {
             return ObdGatewayService.this;
         }
     }
+    public static void saveLogcatToFile(Context context) {
+        Intent emailIntent = new Intent(Intent.ACTION_SEND);
+        emailIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+                emailIntent.setType("text/plain");
+        emailIntent.putExtra(Intent.EXTRA_EMAIL, new String[] {"shaun@jvn.sx"}); //Change this to the correct email
```
Comment (pires) [inline src/main/java/com/github/pires/obd/reader/io/ObdGatewayService.java]: This must be configurable.

File: src/main/java/com/github/pires/obd/reader/io/ObdGatewayService.java [hunk 2]
```diff
@@ -274,5 +274,38 @@ public ObdGatewayService getService() {
...
+        emailIntent.putExtra(Intent.EXTRA_SUBJECT, "Hazmat Debug Logs");
```
Comment (pires) [inline src/main/java/com/github/pires/obd/reader/io/ObdGatewayService.java]: What is Hazmat?

File: src/main/java/com/github/pires/obd/reader/io/ObdGatewayService.java [hunk 3]
```diff
@@ -274,5 +274,38 @@ public ObdGatewayService getService() {
...
+        
+        StringBuilder sb = new StringBuilder();
+        sb.append("\nManufacturer: ").append(Build.MANUFACTURER);
+        sb.append("\nModel: ").append(Build.MODEL);
+        sb.append("\nRelease: ").append(Build.VERSION.RELEASE);
+
+        emailIntent.putExtra(Intent.EXTRA_TEXT, sb.toString());
+
+        String fileName = "hazmat_logcat_"+System.currentTimeMillis()+".txt";
```
Comment (pires) [inline src/main/java/com/github/pires/obd/reader/io/ObdGatewayService.java]: hazmat?

**Comment by sytolk** [comment]
@Pho3niX90 I`m open source programmer. If its have open source library for debugging (like ACRA) I will use it. If its have somethings wrong with this library I will make a PR but to start writing code from the  beginning.. this is not my style :)

**Comment by Pho3niX90** [comment]
@sytolk If you do not write new code how are you a developer? I do not understand?
@pires I have changed the name from my test app to the correct "OBD2 Reader". 
What do you suggest around the email address? This should be the main developers email address, but give the end user the option to change that address?

**Comment by sytolk** [comment]
@Pho3niX90 think about this and maybe you will understand ;)

**Comment by pires** [comment]
@sytolk @Pho3niX90 let's agree to disagree here.

I am fine with native Android crash handling that sends `logcat` logs by email. I am also fine with adding Acra or any other external service that a user sees fit for the same purpose.
What I'm not OK with, and both with this PR and #138 is the lack of configurability by the user.

Please, make all variables configurable, i.e. usernames, passwords, email destination addresses, service URLs and tokens, etc. Then, remove all references to other software you may have written before that is in no way related to Android OBD Reader app. And finally, meld (squash) your commits into a single one and I'll be ready to merge it.

**Comment by Pho3niX90** [comment]
@pires  Is this sufficient?

**Comment by pires** [comment]
@Pho3niX90 looks sharp. Now, I need to crash the app to test.

Meanwhile, can you meld your commits into a single one? Thanks!
